addon/mixins/copyable.js

```diff
@@ -3,14 +3,22 @@ import DS from 'ember-data';
 
 export default Ember.Mixin.create({
   copyable: true,
-  copy: function(options) {
+  copy: function(options, copied) {
     options = options || {};
+    copied = copied || {};
 
     var _this = this;
     return new Ember.RSVP.Promise(function(resolve) {
 
       var model = _this.constructor;
+
+      var id = _this.get('id');
```

Comment:
Using the model's `id` alone as a key into the `copied` object can cause a crash if you have an `id` value re-used across different related objects.

This can be fixed by replacing this line with:
```js
      var id = `${model.modelName}--${_this.get('id')}`;
```